assets/lokomotive-kubernetes/bootkube/resources/charts/kube-apiserver/templates/kube-apiserver.yaml

```diff
@@ -81,6 +109,67 @@ spec:
         - name: ssl-certs-host
           mountPath: /etc/ssl/certs
           readOnly: true
+        - name: data
+          mountPath: /run/kube-apiserver
+      - name: haproxy
+        image: haproxy:2.1.3-alpine
+        volumeMounts:
+        - name: data
+          mountPath: /run/kube-apiserver
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -x && \
+          export ADDRESS=$(cat /run/kube-apiserver/address) && \
+          if [ -z $ADDRESS ]; then \
+            echo "ADDRESS not found" && \
+            exit 1; \
+          fi && \
+          echo "defaults \
+            # Do TLS passthrough \
+            mode tcp \
+            # Required values for both frontend and backend
+            timeout connect 5000ms
+            timeout client 50000ms
+            timeout server 50000ms
+
+          frontend kube-apiserver-internal
+            bind $POD_IP:{{ template "port" . }}
+            default_backend kube-apiserver
+
+          {{- if .Values.apiserver.exposeOnAllPorts }}
+          frontend kube-apiserver-external
+            bind 0.0.0.0:6443
+            default_backend kube-apiserver
+
+          {{- end }}
+          backend kube-apiserver
+            server 1 $ADDRESS:{{ template "port" . }} check" > /run/kube-apiserver/haproxy.cfg && \
+          echo "Connecting to $ADDRESS:{{ template "port" . }}" && \
+          until nc -zv $ADDRESS {{ template "port" . }}; do sleep 1; done && \
+          echo "Connected" && \
+          # From https://github.com/docker-library/haproxy/blob/master/Dockerfile-debian.template#L70
+          exec haproxy -f /run/kube-apiserver/haproxy.cfg
+        env:
+        - name: POD_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        readinessProbe:
+          tcpSocket:
+            port: {{ template "port" . }}
+      initContainers:
+      - name: port-generator
+        image: haproxy:2.1.3-alpine
+        command:
+        - /bin/sh
+        - -c
+        - |
+          echo "127.$(shuf -i 0-255 -n 1).$(shuf -i 0-255 -n 1).$(shuf -i 1-253 -n 1)" | tee /run/kube-apiserver/address
+        volumeMounts:
+        - name: data
+          mountPath: /run/kube-apiserver
```

Comment:
If we decide to keep this sidecar container, I'd consider including it in the yaml only for the single node case. Having it doormant in the yaml, for the HA control plane case (the most common for production), doesn't seem like a good idea IMHO. I think an HA 3 controllers cluster is a very important setup, I'd not slip unneeded containers and bigger attack surface.

Or am I missing something and is not doormant?

Reply:
Removed from the HA case, closing :)